### PR TITLE
network: check specified address settings in more detail

### PR DESCRIPTION
**Pull request by yuwata**
Split-out of #28325.

### CODE REVIEW

**Comment by yuwata** [comment]
All CI failures are unrelated.
CentOS CI: TEST-50-DISSECT,
jammy-amd64: TEST-55-OOMD,
jammy-ppc64el: TEST-36-NUMAPOLICY.

File: src/network/networkd-address.c [hunk 1]
```diff
@@ -2136,6 +2136,12 @@ static int address_section_verify(Address *address) {
                                          address->section->filename, address->section->line);
         }
 
+        if (address->family == AF_INET6 && !socket_ipv6_is_supported())
+                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                         "%s: an IPv6 address configured, but the kernel does not support IPv6. "
```
Comment (bluca) [inline src/network/networkd-address.c]: "was configured"

File: src/network/networkd-address.c [hunk 2]
```diff
@@ -2179,6 +2212,19 @@ static int address_section_verify(Address *address) {
             !FLAGS_SET(address->duplicate_address_detection, ADDRESS_FAMILY_IPV6))
                 address->flags |= IFA_F_NODAD;
 
+        uint32_t filtered_flags = address->family == AF_INET ?
+                address->flags & KNOWN_FLAGS & ~UNMANAGED_FLAGS & ~IPV6ONLY_FLAGS :
+                address->flags & KNOWN_FLAGS & ~UNMANAGED_FLAGS;
+        if (address->flags != filtered_flags) {
```
Comment (bluca) [inline src/network/networkd-address.c:2218]: doesn't this mean that every time the kernel is updated, networkd needs to be updated too? Isn't that a bit too unflexible?

Comment (yuwata) [reply src/network/networkd-address.c:2218]: This verifies [Address] settings in .network files, not verifying the addresses assigned to the system.
So, unless we add more knobs to configure flags, the logic is not necessary to be updated, even if the future kernel adds more flags.

**Comment by yuwata** [comment]
@bluca Thank you the review. The log messages are updated. PTAL.

**Review by bluca — APPROVED** [review]

**Pull request merged**